### PR TITLE
Im4k2 ppm

### PR DESCRIPTION
**Pull request by jozamudi**
## Description

Changed IM4k2 FB_PPM_PowerMeter_IM4K2 settings to match expected value ~10mv

## Motivation and Context

https://jira.slac.stanford.edu/browse/ECS-5463

## How Has This Been Tested?

Philip Heimann has validated the value.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive comments
- [x] Test suite passes locally
- [x] Libraries are set to fixed versions and not ``Always Newest``
- [x] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

**Comment by ZLLentz** [comment]
Can you explain why this works? I read this calculation as the following:
```
scale = 2^23-1 * (10 - -10) =  8388588
real = (raw / 8388588 + -10) * 1 + 8
# sanity check: raw = 0 should be 0V
real = ( 0 - 10 ) + 8 = -2 mV
# value from the manual: 7F FF FF FF -> 10V
real = (2^23-1 / 8388588 + -10) + 8 = -0.999 mV
```
I don't understand the values selected.

**Comment by jozamudi** [comment]
I was taking a deeper look at the values that the terminal is reading.  I don't believe the terminal output value from the table is being used by the terminal CoE (6000:11) value. 

Logging in into the PLC the iRaw Value reding in has a value in ~5100000, if the value from the table was used, the max value that fits the 23 bits is 8388607. If the expected reading is ~10mv, the terminal would have to encode ~9.99v in the ~300000 left in the 23 bits.

The CoE (6000:11) value uses a 32 bit DINT variable. If we use the 32bits for the iTermBits value for the scaling calculations we receive a better result using the following equation.

```
rScale := EXPT(2, iTermBits) - 1; 
fReal := (iRaw / fScale) * fTermMax;
fReal := fReal * fResolution + fOffset;
```

Where:
	iRaw, value from terminal
	 iTermBits, is the bits used by the Value (6000:11) data type.
	 fTermMax, is the highest value in volts
	 fResolution, is the units to display the value. ie 1 = Volt, 1000 = mV

Some sanity checks:
```
iRaw = 0
iTermBits = 32
fTermMax = 10
fResolution = 1

rScale := EXPT(2, 32) - 1; 
4294967296
fReal := (0 / 4294967296) * 10;
0
fReal := 0 * 1 + fOffset;
0 + Offset
```

```
iRaw = 2^32 - 1
iTermBits = 32
fTermMax = 10
fResolution = 1

rScale := EXPT(2, 32) - 1; 
4294967296
fReal := (4294967296 / 4294967296) * 10;
10
fReal := 0 * 1 + fOffset;
10 + Offset
```

A data point from PLC Value.
```
iRaw = 5052416
iTermBits = 32
fTermMax = 10
fResolution = 1000 (V -> mV)

rScale := EXPT(2, 32) - 1; 
4294967296
fReal := (5052416 / 4294967296) * 10;
0.011763572692871094  (V)
fReal := 0.011763572692871094 * 1000 + fOffset;
11.763572692871094 + Offset (mv)
```

**Comment by ZLLentz** [comment]
I see, the CoE value is not the value from the manual. With that knowledge and your most recent commit, your new parameterization makes much, much more sense.

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/IM4K2/FB_PPM_PowerMeter_IM4K2.TcPOU [hunk 1]
```diff
@@ -81,9 +81,11 @@ END_VAR
 // Convert the terminal's integer into a value in millivolts
 fbGetPMVoltage(
     iRaw := iVoltageDINT,
-    iTermBits := 23,
-    fTermMax := 10000,
-    fTermMin := 0,  //For 24-bit to spread 10,000mV across 23 bits
+    iTermBits := 32, // DINT IRaw # of Bit for scaling
+    fTermMax := 10,
+    fTermMin := -10,
+    fOffset := -2, // Add Offset based on fResolution (2mV in this case).
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/IM4K2/FB_PPM_PowerMeter_IM4K2.TcPOU:87]: This is the only small detail that I don't understand. Is this a device-specific calibration offset or a general offset we'd want to apply to all of the high-bit-depth PPMs?

Comment (ZLLentz) [reply plc-kfe-rix-motion/kfe_rix_motion/POUs/IM4K2/FB_PPM_PowerMeter_IM4K2.TcPOU:87]: Or is this a correction on the raw voltage scaling?

Comment (ZLLentz) [reply plc-kfe-rix-motion/kfe_rix_motion/POUs/IM4K2/FB_PPM_PowerMeter_IM4K2.TcPOU:87]: Ok, seems fine. I can believe that there is a small voltage difference caused by the electronics. In general I'm not sure how much anyone is going to notice a 2mV offset.

In the end, with or without this, the final reading is going to be comparing the with-beam voltage to the without-beam voltage so these terms cancel out anyway.

**Review by ZLLentz — APPROVED** [review]
Looks good! Thank you for the investigation and the fix!

**Pull request merged**